src/utils/listener-registry.js

```diff
@@ -65,11 +65,11 @@ ListenerRegistry.prototype.addListener = function( socketWrapper, message ) {
     return;
   }
   
-  if( !this._subscriptionRegistry.isSubscriber( socketWrapper ) ) {
```

Comment:
So this fired the close event in the listener registry before the subscription registry, but the listener in this class requires the subscription registry to have first removed the subscription. Hence the move